backend/danswer/db/engine.py

```diff
@@ -211,25 +230,102 @@ def get_sqlalchemy_async_engine() -> AsyncEngine:
     return _ASYNC_ENGINE
 
 
-def get_session_context_manager() -> ContextManager[Session]:
-    return contextlib.contextmanager(get_session)()
-
-
-def get_session() -> Generator[Session, None, None]:
-    # The line below was added to monitor the latency caused by Postgres connections
-    # during API calls.
-    # with tracer.trace("db.get_session"):
-    with Session(get_sqlalchemy_engine(), expire_on_commit=False) as session:
+# Context variable to store the current tenant ID
+current_tenant_id = contextvars.ContextVar(
```

Comment:
I would add a comment on how this works in an async and sync context just for clarity and future dev work